mlflow/store/_unity_catalog/registry/rest_store.py

```diff
@@ -315,7 +315,14 @@ def _get_workspace_id(self, run_id):
         response = http_request(
             host_creds=host_creds, endpoint=endpoint, method=method, params={"run_id": run_id}
         )
-        response = verify_rest_response(response, endpoint)
+        try:
+            verify_rest_response(response, endpoint)
+        except MlflowException:
+            _logger.warning(
+                f"Unable to fetch model version's source run (with ID {run_id}) "
+                "from tracking server. The source run may be deleted or inaccessible to the "
+                "current user. No run link will be recorded for the model version."
+            )
```

Comment:
```suggestion
            )
            return
```

Should we exit early here?

Reply:
Yes good call to return early!